src/App.tsx

```diff
@@ -15,36 +16,115 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Alphabetically,
+  Length,
+  Reset,
+}
+
+export function sortGoods(
+  goods: string[],
+  type: SortType,
+  reverse: boolean,
+): string[] {
+  const preparedGoods = [...goods];
+
+  switch (type) {
+    case SortType.Alphabetically:
+      preparedGoods.sort((goodA, goodB) => goodA.localeCompare(goodB));
+      break;
+    case SortType.Length:
+      preparedGoods.sort((goodA, goodB) => goodA.length - goodB.length);
+      break;
+    default:
+      break;
+  }
+
+  if (reverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [goods, setGoods] = useState<string[]>(goodsFromServer);
+  const [sort, setSort] = useState<SortType | null>(null);
+  const [reverse, setReverse] = useState<boolean>(false);
+
+  const handleSort = (type: SortType) => {
+    const sortedGoods = sortGoods(goodsFromServer, type, reverse);
+
+    setSort(type);
+    setGoods(sortedGoods);
+  };
+
+  const handleReverse = () => {
+    const newReverse = !reverse;
+    const sortedGoods = sortGoods(
+      goodsFromServer,
+      sort ?? SortType.Reset,
+      newReverse,
+    );
+
+    setReverse(newReverse);
+    setGoods(sortedGoods);
+  };
+
+  const handleReset = () => {
+    setGoods(goodsFromServer);
+    setSort(null);
+    setReverse(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          onClick={() => handleSort(SortType.Alphabetically)}
+          className={cn('button is-info', {
+            'is-light': sort !== SortType.Alphabetically,
+          })}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          onClick={() => handleSort(SortType.Length)}
+          className={cn('button is-success', {
+            'is-light': sort !== SortType.Length,
+          })}
```

Comment:
To improve UX, consider adding an `aria-pressed` attribute to the button to indicate when it's active. For example: `aria-pressed={sort === SortType.Length}`.